unix/Makefile

```diff
@@ -108,6 +114,19 @@ SRC_MOD += modjni.c
 endif
 
 # source files
+ifeq ($(TARGET), freedos)
+# no moduselect for freedos
+SRC_C = \
+	main.c \
+	gccollect.c \
+	unix_mphal.c \
+	input.c \
+	file.c \
+	modos.c \
+	alloc.c \
+	coverage.c \
+	$(SRC_MOD)
+else
```

Comment:
You should do SRC_C =  to specify common modules and then use SRC_C += inside an ifneq to freedos to add moduselect